CAE_datacleaning/g032_cae_phys.ipynb

```diff
@@ -0,0 +1,491 @@
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import torch\n",
+    "import torch.nn as nn\n",
+    "import torch.nn.functional as F\n",
+    "import torch.optim as optim\n",
+    "from torch.utils.data import DataLoader,TensorDataset\n",
+    "from pytorch_model_summary import summary\n",
+    "from tsnecuda import TSNE\n",
+    "#from sklearn.preprocessing import StandardScaler"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#check for GPU\n",
+    "DEVICE = torch.device(\"cuda:0\" if torch.cuda.is_available() else \"cpu\")\n",
+    "print(DEVICE)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# .npy files extracted from file: /global/project/projectdirs/m2676/data/lngs/pgt/raw/geds/LPGTA_r0030_20200718T161311Z_phys_geds_raw.lh5\n",
+    "# have to process raw file using raw_to_dsp.py from pygama first, then save waveforms and other parameters into npy file\n",
+    "\n",
+    "#load data and print dimensions of data set\n",
+    "import matplotlib.pyplot as plt\n",
+    "import numpy as np\n",
+    "\n",
+    "waveforms = np.load('g032_waveforms_phys.npy')\n",
+    "psd_sum10 = np.load('g032_psd_sum10_phys.npy')\n",
+    "bl_slope  = np.load('g032_bl_slope_phys.npy')\n",
+    "\n",
+    "train_data = waveforms[0:7650,:]\n",
+    "test_data = waveforms[7650:9600,:] #lose 6 observations for validation\n",
+    "test_psd10 = psd_sum10[7650:9600]\n",
+    "test_bl_slope = bl_slope[7650:9600]\n",
+    "\n",
+    "print(waveforms.shape)\n",
+    "print(train_data.shape)\n",
+    "print(test_data.shape)\n",
+    "print(test_psd10.shape)\n",
+    "print(test_bl_slope.shape)\n",
+    "\n",
+    "plt.plot(np.arange(len(waveforms[44,:])), waveforms[44,:]) #anomalous sample wf \n",
+    "plt.show()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#Preprocess and convert data to tensors\n",
+    "\n",
+    "#scaler = StandardScaler()\n",
+    "#scaled_data = scaler.fit_transform(train_data,y=None) #scales train data \n",
+    "#print(scaled_data.shape)\n",
+    "\n",
+    "train_data, test_data = map(torch.tensor, (train_data, test_data))\n",
+    "\n",
+    "# batch size \n",
+    "bs = 50 \n",
+    "\n",
+    "#train data loader\n",
+    "train_ds = TensorDataset(train_data)\n",
+    "train_dl = DataLoader(train_ds, batch_size=bs) #removed batch shuffling here\n",
+    "\n",
+    "#test data loader\n",
+    "valid_ds = TensorDataset(test_data)\n",
+    "valid_dl = DataLoader(valid_ds, batch_size=bs)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#Send data to GPU\n",
+    "def preprocess(x):\n",
+    "    return x.to(DEVICE)\n",
+    "\n",
+    "class WrappedDataLoader:\n",
```

Comment:
You don't need to send data to GPU one by one. usually what I do is during training, I send each batch of data to GPU, like this:
for i, (images, labels) in enumerate(train_loader):
            classifier.train()
            images = images.to(DEVICE)
            labels = labels.view(-1,1)
            labels = labels.to(DEVICE)
This might be slightly slower, but it saves the memory on GPU.